File: src/main/java/cn/hselfweb/mes/arithmetic/entity/GAList.java
package cn.hselfweb.mes.arithmetic.entity;

import cn.hselfweb.mes.enity.CraftExtend;
import com.google.gson.Gson;
import com.google.gson.stream.JsonReader;

import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.List;

public class GAList {
    private static final String dataPath = "GAData.json";
    private int DeviceNumber;
    private List<List<CraftExtend>> list_sum = new ArrayList<List<CraftExtend>>();

    public int getDeviceNumber() {
        return DeviceNumber;
    }

    public void setDeviceNumber(int deviceNumber) {
        DeviceNumber = deviceNumber;
    }

    public List<List<CraftExtend>> getList_sum() {
        return list_sum;
    }

    public void setList_sum(List<List<CraftExtend>> list_sum) {
        this.list_sum = list_sum;
    }

    /**
     * 实例化  调度工序列表
     *
     * @throws FileNotFoundException
     */
    public static final GAList Load() throws FileNotFoundException {
        Gson gson = new Gson();
        JsonReader reader = new JsonReader(new InputStreamReader(new FileInputStream(dataPath)));
        GAList ListC = gson.fromJson(reader, GAList.class);
        int gjCount = ListC.getList_sum().size();
        int gxCount = ListC.getList_sum().get(0).size();
        for (int i = 0; i < gjCount; i++) {

            for (int j = 0; j < gxCount; j++) {
                ListC.getList_sum().get(i).get(j).setGid(j + 1);
                ListC.getList_sum().get(i).get(j).setWpid(i + 1);
                if (j != 0) {
                    //设置前序工序
                    ListC.getList_sum().get(i).get(j).setExtendCraftByPreEcId(ListC.getList_sum().get(i).get(j - 1));
                }
                if (j < gxCount - 1) {
                    //设置后序工序
                    ListC.getList_sum().get(i).get(j).setExtendCraftByAftEcId(ListC.getList_sum().get(i).get(j + 1));
                }

            }
        }
        return ListC;
    }

}

File: src/main/java/cn/hselfweb/mes/arithmetic/entity/GAList.kt
package cn.hselfweb.mes.arithmetic.entity

import cn.hselfweb.mes.enity.CraftExtend
import com.google.gson.Gson
import com.google.gson.stream.JsonReader
import java.io.FileInputStream
import java.io.FileNotFoundException
import java.io.InputStreamReader
import java.util.*

class GAList {
    var DeviceNumber: Int = 0
    var list_sum: List<List<CraftExtend>> = ArrayList()

    companion object {
        private val dataPath = "GAData.json"
        /**
         * 实例化  调度工序列表
         * @throws FileNotFoundException
         */
        @Throws(FileNotFoundException::class)
        fun Load(): GAList {
            val gson = Gson()
            val reader = JsonReader(InputStreamReader(FileInputStream(dataPath)))
            val ListC = gson.fromJson<GAList>(reader, GAList::class.java)
            val gjCount = ListC.list_sum.size
            val gxCount = ListC.list_sum[0].size
            for (i in 0 until gjCount) {
                for (j in 0 until gxCount) {
                    ListC.list_sum[i][j].gid = j + 1
                    ListC.list_sum[i][j].wpid = (i + 1).toLong()
                    if (j != 0) {
                        //设置前序工序
                        ListC.list_sum[i][j].extendCraftByPreEcId = ListC.list_sum[i][j - 1]
                    }
                    if (j < gxCount - 1) {
                        //设置后序工序
                        ListC.list_sum[i][j].extendCraftByAftEcId = ListC.list_sum[i][j + 1]
                    }

                }
            }
            return ListC
        }
    }

}
